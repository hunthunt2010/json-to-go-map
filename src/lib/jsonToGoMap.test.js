import jsonToMap from "./jsonToGoMap";
describe("the jsonToGoMap converter", () => {
    test("when an empty value is given, it should return an empty string", () => {
        const result = jsonToMap("");
        expect(result).toEqual("");
    });
    
    test("when an invalid JSON is given, it should return an error mensage", () => {
        const result = jsonToMap(`{"test": invalid`);
        expect(result).toEqual("Unexpected token i in JSON at position 9");
    });

    test("when a simple key/value object is given, it should return a valid result", () => {
        const result = jsonToMap(`{"key": "value"}`);
        expect(result).toMatchSnapshot();
    });

    test("when null value is given, it should return a valid result", () => {
        const result = jsonToMap(`{"key": null}`);
        expect(result).toMatchSnapshot();
    });

    test("when array is given, it should return an interface list", () => {
        const result = jsonToMap(`["A", "B", 1]`);
        expect(result).toMatchSnapshot();
    });

    test("when a complex object is given", () => {
        const result = jsonToMap(`{"array":["A", "B",1],"foo":{"bar":0.5}}`);
        expect(result).toMatchSnapshot();
    });
});
