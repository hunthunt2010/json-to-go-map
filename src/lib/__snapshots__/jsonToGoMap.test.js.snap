// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the jsonToGoMap converter when a complex object is given 1`] = `
"map[string]interface{}{
	\\"array\\": []interface{}{
		\\"A\\",
		\\"B\\",
		1,
	},
	\\"foo\\": map[string]interface{}{
		\\"bar\\": 0.5,
	},
}"
`;

exports[`the jsonToGoMap converter when a simple key/value object is given, it should return a valid result 1`] = `
"map[string]interface{}{
	\\"key\\": \\"value\\",
}"
`;

exports[`the jsonToGoMap converter when array is given, it should return an interface list 1`] = `
"[]interface{}{
	\\"A\\",
	\\"B\\",
	1,
}"
`;

exports[`the jsonToGoMap converter when null value is given, it should return a valid result 1`] = `
"map[string]interface{}{
	\\"key\\": nil,
}"
`;
